persist/fs/retriever.go

```diff
@@ -348,15 +348,16 @@ func (r *blockRetriever) Stream(
 	ctx context.Context,
 	shard uint32,
 	id ident.ID,
-	startTime time.Time,
+	startTime, endTime time.Time,
```

Comment:
Instead of requiring `endTime` being passed here (which callers could get wrong, or just not have access to potentially) can you calculate from the namespace retention options?

i.e. some variation of
```go
req.end = startTime.Add(r.nsMetadata.Options().RetentionOptions().BlockSize())
```

We could cache the block size on the retriever too if we wanted to avoid the method calls, *shrug*.